coral-component-colorpicker/src/scripts/ColorPicker.js

```diff
@@ -0,0 +1,368 @@
+      'key:down ._coral-ColorPicker-input:not([readonly])': '_onKeyDown',
+      'key:down [handle="colorPreview"]': '_onKeyDown',
+      'click [handle="colorPreview"]': '_onColorPreviewClick',
+      'key:esc input': '_onKeyEsc',
+      'key:enter input': '_onKeyEsc',
+      'capture:change  [handle="input"]': '_onColorInputChange',
+      'change [handle="propertiesView"]': '_onPropertyChange'
+    });
+
+    // Overlay
+    events[`global:capture:coral-overlay:beforeopen #${overlayId}`] = '_beforeOverlayOpen';
+    events[`global:capture:coral-overlay:close #${overlayId}`] = '_onOverlayClose';
+    events[`global:key:esc #${overlayId}`] = '_onKeyEsc';
+
+    // Events
+    this._delegateEvents(events);
+    this.value = "";
+    this._format = ColorFormats.HSL;
+    this._sendChaneEvent = this._chaneEventDebounced();
+  }
+    
+  /** @ignore */  
+  connectedCallback() {
+    super.connectedCallback();
+
+    const overlay = this._elements.overlay;
+    // Cannot be open by default when rendered
+    overlay.removeAttribute('open');
+    // Restore in DOM
+    if (overlay._parent) {
+      overlay._parent.appendChild(overlay);
+    }
+  }
+
+  /** @ignore */
+  disconnectedCallback() {
+    super.disconnectedCallback();
+
+    const overlay = this._elements.overlay;
+    // In case it was moved out don't forget to remove it
+    if (!this.contains(overlay)) {
+      overlay._parent = overlay._repositioned ? document.body : this;
+      overlay.remove();
+    }
+  }
+  
+  /** @ignore */
+  render() {
+    super.render();
+
+    this.classList.add(CLASSNAME);
+
+    this.setAttribute('role', 'group');
+
+    const frag = document.createDocumentFragment();
+
+    // Render template
+    frag.appendChild(this._elements.input);
+    frag.appendChild(this._elements.buttonWrapper);
+    frag.appendChild(this._elements.overlay);
+
+    // Support cloneNode
+    while (this.firstChild) {
+      const child = this.firstChild;
+      if (child.nodeType === Node.ELEMENT_NODE && child.hasAttribute('handle')) {
+        this.removeChild(child);
+      } 
+      else {
+        frag.appendChild(child);
+      }
+    }
+
+    this.appendChild(frag);
+
+    // These should be used to set a property since property handler aren't called until elements are attached to dom.
+    // Attribute values are delivered to change-listeners even if element isn't attached to dom yet, so attributes 
+    // can be set to e.g. this._elements.colorPreview.
+    this._input = this.querySelector("[handle='input']");
+    this._preview = this.querySelector("[handle='colorPreview']");
+    this._overlay = this.querySelector("[handle='overlay']");
+    this._overlay.setAttribute('aria-live', 'off');
+    this._properties = this._overlay.querySelector("[handle='propertiesView']");
+    this._update(this._value);
+  }
+
+  /** @ignore */
+  static get observedAttributes() {
+    return super.observedAttributes.concat([
+      'value',
+      'formats',
```

Comment:
Shouldn't it just be called `format` as in most cases a user would only be choosing a single format?

Reply:
there might be case to allow a set oof fomats.